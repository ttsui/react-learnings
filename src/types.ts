export interface Address {
  street: string[];
  suburb: string;
  postcode?: number;
  state: string;
}