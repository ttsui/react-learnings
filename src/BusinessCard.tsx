import React from "react";
import { BusinessCardInfo } from "./types";

export const BusinessCard: React.FC<BusinessCardInfo> = ({ address, phoneNumber }) => (
  <section
    style={{
      backgroundColor: "#fef5f0",
      color: "darkgray",

      fontFamily: "sans-serif",
      fontSize: "13pt",
      fontWeight: "bold",
      fontVariant: "all-small-caps",

      display: "flex",
      flexDirection: "column",
      justifyContent: "space-between",

      boxSizing: "border-box",
      padding: "30px",
      width: "530px",
      height: "317px"
    }}
  >
    <section
      style={{
        display: "flex",
        alignItems: "flex-start",
        flexDirection: "column"
      }}
    >
      <h1
        style={{
          color: "#59504b",
          fontSize: "25pt",
          marginBlockEnd: 0,
          marginBlockStart: 0
        }}
      />
    </section>

    <section
      style={{
        display: "flex",
        alignItems: "flex-end",
        flexDirection: "column",

        color: "darkgray",

        fontFamily: "sans-serif",
        fontSize: "13pt",
        fontWeight: "bold",
        fontVariant: "all-small-caps"
      }}
    >
      <div>Call: {phoneNumber}</div>
      <div>Visit: {address.street.concat(address.suburb, address.state, String(address.postcode)).join(", ")}</div>
    </section>
  </section>
);
