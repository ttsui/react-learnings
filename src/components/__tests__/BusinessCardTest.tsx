import React from "react";
import { render } from "react-testing-library";
import { BusinessCard } from "../BusinessCard";
import { BusinessCardInfo } from "../../types";

it("renders", () => {
  const cardInfo: BusinessCardInfo = {
    name: "Mariana Napolitani",
    title: "Broker",
    address: {
      street: ["Unit 5", "23 Market St"],
      suburb: "Sydney",
      state: "NSW",
      postcode: 2000
    },
    phoneNumber: "(02) 9804 11234",
    email: "mariana@myestate.com",
    website: "www.myestate.com"
  };

  const { container } = render(<BusinessCard {...cardInfo} />);

  expect(container).toMatchSnapshot();
});
