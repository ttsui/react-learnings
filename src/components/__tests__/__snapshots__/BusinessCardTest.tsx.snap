// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders welcome message 1`] = `
<div>
  <section
    style="background-color: rgb(254, 245, 240); border-radius: 5px; box-shadow: 0 2px 1px rgba(0,0,0,.08); color: darkgray; font-family: sans-serif; font-size: 0.9em; font-weight: bold; font-variant: all-small-caps; display: flex; flex-direction: column; justify-content: space-between; box-sizing: border-box; padding: 1.5em; margin: 20px; width: 420px; height: 240px;"
  >
    <section
      style="display: flex; align-items: flex-start; flex-direction: column;"
    >
      <h1
        style="color: rgb(89, 80, 75); font-size: 1.8em; margin-block-end: 0; margin-block-start: 0;"
      >
        Mariana Napolitani
      </h1>
      <div>
        Broker
      </div>
    </section>
    <section
      style="display: flex; align-items: flex-end; flex-direction: column;"
    >
      <div>
        Call: 
        (02) 9804 11234
      </div>
      <div>
        Visit: 
        Unit 5, 23 Market St, Sydney, NSW, 2000
      </div>
      <div>
        Email: 
        <span
          style="font-variant: unset;"
        >
          mariana@myestate.com
        </span>
      </div>
      <div>
        Click: 
        www.myestate.com
      </div>
    </section>
  </section>
</div>
`;
